addons/pos_details/static/src/js/reports.js

```diff
@@ -0,0 +1,84 @@
+odoo.define('pos_details.pos_details', function (require) {
+"use strict";
+
+var PosBaseWidget = require('point_of_sale.BaseWidget');
+var devices = require('point_of_sale.devices');
+var screens = require('point_of_sale.screens');
+var chrome = require('point_of_sale.chrome');
+var core = require('web.core');
+var Model = require('web.DataModel');
+
+var QWeb = core.qweb;
+var _t = core._t;
+
+
+var SaleDetailsButton = PosBaseWidget.extend({
+    template: 'SaleDetailsButton',
+    start: function(){
+        var self = this;
+        this.$el.click(function(){
+            self.pos.proxy.print_sale_details();
+        });
+    },
+});
+
+chrome.Chrome.include({
+    build_widgets: function () {
+
+        // add report widget to left of proxy widget
+        var proxy_status_index = _.findIndex(this.widgets, function (widget) {
+            return widget.name === "proxy_status";
+        });
+
+        this.widgets.splice(proxy_status_index, 0, {
+                'name': 'sale_details',
+                'widget': SaleDetailsButton,
+                'append': '.pos-rightheader',
+            });
+
+            this._super();
+        },
+});
+
+devices.ProxyDevice.include({
+    print_sale_details: function() { 
+        var self = this;
+        new Model('pos.order').call('get_sale_details').then(function(result){
+            var env = {
+                company: self.pos.company,
+                pos: self.pos,
+                products: result.products,
+                payments: result.payments,
+                taxes: result.taxes,
+                total_paid: result.total_paid,
+                date: (new Date()).toLocaleString(),
+            };
+            var report = QWeb.render('SaleDetailsReport', env);
+
+            // self.receipt_queue.push(report);
+            
+            function send_printing_job(){
+                if (self.receipt_queue.length > 0){
+                    var r = self.receipt_queue.shift();
+                    self.message('print_xml_receipt',{ receipt: r },{ timeout: 5000 })
+                        .then(function(){
+                            send_printing_job();
+                        },function(error){
+                            if (error) {
+                                self.pos.chrome.gui.show_popup('error-traceback',{
+                                    'title': _t('Printing Error: ') + error.data.message,
+                                    'body':  error.data.debug,
+                                });
+                                return;
+                            }
+                            self.receipt_queue.unshift(r);
+                        });
+                }
+            }
+            send_printing_job();
```

Comment:
I think we should be able to use the `print_receipt` function in `point_of_sale.devices`.